include/clap/ext/draft/location.h

```diff
@@ -0,0 +1,73 @@
+#pragma once
+
+#include "../../color.h"
+#include "../../plugin.h"
+#include "../../string-sizes.h"
+
+// This extension allows a host to tell the plugin more about its position
+// within a project or session.
+
+static CLAP_CONSTEXPR const char CLAP_EXT_LOCATION[] = "clap.location/1";
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+enum {
+   // Represents a document/project/session.
+   CLAP_PLUGIN_LOCATION_PROJECT = 1,
+
+   // Represents a group of tracks.
+   // It can contain both track groups, tracks and devices (post processing).
+   // The first device within a track group has the index of
+   // the last track or track group within this group + 1.
```

Comment:
Specifically the first *device*, or should this be the first *child* (whether that is track group, track, or device?).

What is the purpose of setting the index to that value? Is it to tell the plugin the range of location elements that are children (direct or not) of the parent element? That should probably be made clear.

Is the first child required to immediately follow its parent element in the `clap_plugin_location_element_t *path` array in `set_location`?

Reply:
Actually, it could be simplified.
We could say that index starts at 0 for tracks and then 0 again for devices.

This is the model:

```
track_group
 `- track0 -|
 `- track1 -| -> (+) -> device0 -> device1 -> ... 
 `- ...    -|
```